enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -462,15 +479,24 @@ def _capture_user_overrides(self, **kwargs):
         self.kernel_launch_timeout = float(
             env.get("KERNEL_LAUNCH_TIMEOUT", default_kernel_launch_timeout)
         )
-        self.user_overrides.update(
-            {
-                key: value
-                for key, value in env.items()
-                if key.startswith("KERNEL_")
-                or key in self.env_process_whitelist
-                or key in self.env_whitelist
-            }
-        )
+        # kwargs['env'] gets updated with each kernel start / restart.
+        if (
+            not self.user_overrides
+        ):  # user_overrides preserve the original envs with which the kernel is started.
+            self.user_overrides.update(
+                {
+                    key: value
+                    for key, value in env.items()
+                    if key.startswith("KERNEL_")
+                    or key in self.parent.parent.env_process_whitelist
+                    or key in self.parent.parent.env_whitelist
```

Comment:
These should revert to not include `parent.parent` as these are "massaged" in `__init__()` via `self._link_dependent_props()`:
```suggestion
                    or key in self.env_process_whitelist
                    or key in self.env_whitelist
```